File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/C++/s893190600_comments_removed.cpp
// End of Code
#include <iostream>
#include <vector>
#include <map>
#include <set>
#include <stack>
#include <queue>
#include <algorithm>
#include <string>
#include <sstream>
#include <cmath>
#include <unordered_map>
#include <unordered_set>

using namespace std;

class PP {
public:
    int key, val;
    PP(int key, int val) : key(key), val(val) {}
};

class PPP {
public:
    int key, val1, val2;
    PPP(int key, int val1, int val2) : key(key), val1(val1), val2(val2) {}
};

class PPL {
public:
    long long key, val;
    PPL(long long key, long long val) : key(key), val(val) {}
};

class PPDL {
public:
    long long key;
    vector<long long> val;
    PPDL(long long key, vector<long long> val) : key(key), val(val) {}
    void dump() {
        cout << "key = " << key << "  val ";
        for (int i = 0; i < val.size(); i++) {
            cout << "[" << val[i] << "] ";
        }
        cout << endl;
    }
};

class PPKEY {
private:
    int key, val;
public:
    PPKEY(int key, int val) : key(key), val(val) {}
    int getKey() { return key; }
    int getVal() { return val; }
    bool operator==(const PPKEY& other) const {
        return key == other.key && val == other.val;
    }
};

class PPLKEY {
private:
    long long key, val;
public:
    PPLKEY(long long key, long long val) : key(key), val(val) {}
    long long getKey() { return key; }
    long long getVal() { return val; }
    bool operator==(const PPLKEY& other) const {
        return key == other.key && val == other.val;
    }
};

namespace std {
    template <>
    struct hash<PPKEY> {
        size_t operator()(const PPKEY& k) const {
            return hash<int>()(k.getKey()) ^ hash<int>()(k.getVal());
        }
    };

    template <>
    struct hash<PPLKEY> {
        size_t operator()(const PPLKEY& k) const {
            return hash<long long>()(k.getKey()) ^ hash<long long>()(k.getVal());
        }
    };
}

class MyInput {
private:
    istream& in;
    int pos;
    int readLen;
    char buffer[1024 * 8];
    char str[500 * 8 * 2];
    bool isDigit[256];
    bool isSpace[256];
    bool isLineSep[256];

public:
    MyInput(istream& is) : in(is), pos(0), readLen(0) {
        for (int i = 0; i < 10; i++) {
            isDigit['0' + i] = true;
        }
        isDigit['-'] = true;
        isSpace[' '] = isSpace[''] = isSpace['
'] = isSpace['	'] = true;
        isLineSep[''] = isLineSep['
'] = true;
    }

    int read() {
        if (pos >= readLen) {
            pos = 0;
            readLen = in.readsome(buffer, sizeof(buffer));
            if (readLen <= 0) {
                throw runtime_error("End of file");
            }
        }
        return buffer[pos++];
    }

    int nextInt() {
        int len = 0;
        str[len++] = nextChar();
        len = reads(len, isSpace);
        int i = 0;
        int ret = 0;
        if (str[0] == '-') {
            i = 1;
        }
        for (; i < len; i++) ret = ret * 10 + str[i] - '0';
        if (str[0] == '-') {
            ret = -ret;
        }
        return ret;
    }

    long long nextLong() {
        int len = 0;
        str[len++] = nextChar();
        len = reads(len, isSpace);
        int i = 0;
        long long ret = 0L;
        if (str[0] == '-') {
            i = 1;
        }
        for (; i < len; i++) ret = ret * 10 + str[i] - '0';
        if (str[0] == '-') {
            ret = -ret;
        }
        return ret;
    }

    double nextDouble() {
        int len = 0;
        str[len++] = nextChar();
        len = reads(len, isSpace);
        int i = 0;
        double ret = 0;
        if (str[0] == '-') {
            i = 1;
        }
        int cnt = 0;
        for (; i < len; i++) {
            if (str[i] == '.') {
                cnt = 10;
                continue;
            }
            if (cnt == 0) {
                ret = ret * 10 + str[i] - '0';
            } else {
                ret = ret + ((double)(str[i] - '0') / cnt);
                cnt *= 10;
            }
        }
        if (str[0] == '-') {
            ret = -ret;
        }
        return ret;
    }

    string nextString() {
        string ret(nextDChar());
        ret.erase(remove_if(ret.begin(), ret.end(), [](unsigned char c) { return isspace(c); }), ret.end());
        return ret;
    }

    vector<char> nextDChar() {
        int len = 0;
        len = reads(len, isSpace);
        vector<char> ret(len + 1);
        for (int i = 0; i < len; i++) ret[i] = str[i];
        ret[len] = 0x00;
        return ret;
    }

    char nextChar() {
        while (true) {
            int c = read();
            if (!isSpace[c]) {
                return (char)c;
            }
        }
    }

    int reads(int len, bool accept[]) {
        try {
            while (true) {
                int c = read();
                if (accept[c]) {
                    break;
                }
                if (sizeof(str) == len) {
                    vector<char> rep(sizeof(str) * 3 / 2);
                    copy(str, str + sizeof(str), rep.begin());
                    str = rep.data();
                }
                str[len++] = (char)c;
            }
        } catch (const runtime_error& e) {
        }
        return len;
    }
};

class Graph {
private:
    map<int, vector<int>> data;

public:
    void add(int from, int to) {
        data[from].push_back(to);
    }

    void del(int from, int to) {
        auto it = data.find(from);
        if (it != data.end()) {
            it->second.erase(remove(it->second.begin(), it->second.end(), to), it->second.end());
        }
    }

    vector<int> get(int key) {
        return data[key];
    }

    bool contains(int key) {
        return data.find(key) != data.end();
    }

    set<int> keySet() {
        set<int> keys;
        for (const auto& pair : data) {
            keys.insert(pair.first);
        }
        return keys;
    }

    bool isConnect(int key_1, int key_2) {
        auto it = data.find(key_1);
        if (it != data.end()) {
            return find(it->second.begin(), it->second.end(), key_2) != it->second.end();
        }
        return false;
    }

    vector<PP> distList(int key) {
        vector<PP> dist;
        set<int> mark;
        stack<PP> stack;
        stack.push(PP(key, 0));

        while (!stack.empty()) {
            PP wk = stack.top();
            stack.pop();

            int pp = wk.key;
            int dd = wk.val;
            mark.insert(pp);

            dist.push_back(PP(pp, dd));

            vector<int> list = get(pp);

            for (int next : list) {
                if (mark.find(next) != mark.end()) continue;
                stack.push(PP(next, dd + 1));
            }
        }
        return dist;
    }

    vector<int> distV(int key) {
        vector<int> dist(data.size() + 1, -1);
        set<int> mark;
        stack<PP> stack;
        stack.push(PP(key, 0));

        while (!stack.empty()) {
            PP wk = stack.top();
            stack.pop();

            int pp = wk.key;
            int dd = wk.val;
            mark.insert(pp);

            dist[pp] = dd;

            vector<int> list = get(pp);

            for (int next : list) {
                if (mark.find(next) != mark.end()) continue;
                stack.push(PP(next, dd + 1));
            }
        }
        return dist;
    }

    int bridgeDfs(int now, int pre, map<int, int>& mapCnt, map<int, int>& mapLow, set<int>& mark, int& number) {
        mark.insert(now);
        mapCnt[now] = number;
        mapLow[now] = number;
        int low;
        for (int next : get(now)) {
            if (next == pre) continue;
            if (mark.find(next) != mark.end()) {
                if (mapLow[now] > mapLow[next]) {
                    mapLow[now] = mapLow[next];
                }
                continue;
            }
            number++;
            low = bridgeDfs(next, now, mapCnt, mapLow, mark, number);
            if (mapLow[now] > low) {
                mapLow[now] = low;
            }
        }
        return mapLow[now];
    }

    int bridgeCnt(int start) {
        map<int, int> mapCnt;
        map<int, int> mapLow;
        set<int> mark;
        int number = 0;
        bridgeDfs(start, start, mapCnt, mapLow, mark, number);
        int ans = 0;
        for (const auto& pair : mapCnt) {
            if (pair.second == mapLow[pair.first]) {
                ans++;
            }
        }
        return ans - 1;
    }

    void dump() {
        for (const auto& pair : data) {
            cout << pair.first << " : ";
            for (int val : pair.second) {
                cout << val << " ";
            }
            cout << endl;
        }
    }
};

class GraphWith {
private:
    map<int, vector<PP>> data;

public:
    void add(int key, PP p) {
        data[key].push_back(p);
    }

    vector<PP> get(int key) {
        return data[key];
    }

    bool contains(int key) {
        return data.find(key) != data.end();
    }

    bool isConnect(int key_1, int key_2) {
        auto it = data.find(key_1);
        if (it != data.end()) {
            for (const PP& p : it->second) {
                if (p.key == key_2) {
                    return true;
                }
            }
        }
        return false;
    }

    int distance(int key_1, int key_2) {
        set<int> mark;
        stack<PP> stack;
        stack.push(PP(key_1, 0));

        while (!stack.empty()) {
            PP wk = stack.top();

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/Python/s893190600_comments_removed.py
import sys
import math
from collections import deque, defaultdict, Counter, OrderedDict

class Solver:
    def __init__(self, in_stream, out_stream):
        self.in_stream = in_stream
        self.out_stream = out_stream

    def solve(self):
        H = self.ni()
        W = self.ni()
        C = [['.' for _ in range(W + 2)] for _ in range(H + 2)]
        for i in range(H):
            line = self.ns()
            for j in range(W):
                C[i + 1][j + 1] = line[j]
        if H == 1 and W == 1:
            self.prn("No")
            return
        for i in range(1, H + 1):
            for j in range(1, W + 1):
                if C[i][j] == '#' and C[i - 1][j] != '#' and C[i + 1][j] != '#' and C[i][j - 1] != '#' and C[i][j + 1] != '#':
                    self.prn("No")
                    return
        self.prn("Yes")

    def prn(self, s):
        self.out_stream.write(f"{s}\n")

    def prr(self, s):
        self.out_stream.write(s)

    def ni(self):
        return int(self.ns())

    def nl(self):
        return int(self.ns())

    def nd(self):
        return float(self.ns())

    def ns(self):
        return self.in_stream.readline().strip()

    def ndi(self, n):
        return [self.ni() for _ in range(n)]

    def ndl(self, n):
        return [self.nl() for _ in range(n)]

    def ndd(self, n):
        return [self.nd() for _ in range(n)]

    def nds(self, n):
        return [self.ns() for _ in range(n)]

    def nddi(self, n, m):
        return [self.ndi(m) for _ in range(n)]

    def nddl(self, n, m):
        return [self.ndl(m) for _ in range(n)]

    def isRightMin(self, a, f, index, key):
        if f and a[index] >= key:
            return True
        elif not f and a[index] > key:
            return True
        else:
            return False

    def binarySearchRightMin(self, a, f, key):
        ng = -1
        ok = len(a)
        while abs(ok - ng) > 1:
            mid = (ok + ng) // 2
            if self.isRightMin(a, f, mid, key):
                ok = mid
            else:
                ng = mid
        return ok

    def isLeftMax(self, a, f, index, key):
        if f and a[index] <= key:
            return True
        elif not f and a[index] < key:
            return True
        else:
            return False

    def binarySearchLeftMax(self, a, f, key):
        ng = -1
        ok = len(a)
        while abs(ok - ng) > 1:
            mid = (ok + ng) // 2
            if self.isLeftMax(a, f, mid, key):
                ng = mid
            else:
                ok = mid
        return ng

class PP:
    def __init__(self, key, val):
        self.key = key
        self.val = val

    def getKey(self):
        return self.key

    def getVal(self):
        return self.val

class PPP:
    def __init__(self, key, val1, val2):
        self.key = key
        self.val1 = val1
        self.val2 = val2

    def getKey(self):
        return self.key

    def getVal1(self):
        return self.val1

    def getVal2(self):
        return self.val2

class PPL:
    def __init__(self, key, val):
        self.key = key
        self.val = val

    def getKey(self):
        return self.key

    def getVal(self):
        return self.val

class PPDL:
    def __init__(self, key, val):
        self.key = key
        self.val = val

    def getKey(self):
        return self.key

    def getVal(self):
        return self.val

    def dump(self, out_stream):
        out_stream.write(f"key = {self.key}  val ")
        for v in self.val:
            out_stream.write(f"[{v}] ")
        out_stream.write("\n")

class PPKEY:
    def __init__(self, key, val):
        self.key = key
        self.val = val

    def getKey(self):
        return self.key

    def getVal(self):
        return self.val

    def __eq__(self, other):
        if isinstance(other, PPKEY):
            return self.key == other.key and self.val == other.val
        return False

    def __hash__(self):
        return hash((self.key, self.val))

class PPLKEY:
    def __init__(self, key, val):
        self.key = key
        self.val = val

    def getKey(self):
        return self.key

    def getVal(self):
        return self.val

    def __eq__(self, other):
        if isinstance(other, PPKEY):
            return self.key == other.key and self.val == other.val
        return False

    def __hash__(self):
        return hash((self.key, self.val))

class Pair:
    def __init__(self, key, value):
        self.key = key
        self.value = value

    def getKey(self):
        return self.key

    def getValue(self):
        return self.value

class MapCounter:
    def __init__(self, reverse=False):
        if reverse:
            self.map = OrderedDict(sorted({}, reverse=True))
        else:
            self.map = OrderedDict()

    def add(self, key, cnt=1):
        if key in self.map:
            self.map[key] += cnt
        else:
            self.map[key] = cnt

    def remove(self, key):
        self.sub(key, 1, False)

    def sub(self, key, cnt=1, minus=True):
        if key in self.map:
            if self.map[key] > cnt or minus:
                self.map[key] -= cnt
            else:
                del self.map[key]

    def set(self, key, cnt):
        self.map[key] = cnt

    def getCountwithNull(self, key):
        return self.map.get(key)

    def getCount(self, key):
        return self.map.get(key, 0)

    def getKey(self):
        return self.map.keys()

    def getKeyCount(self):
        return len(self.map)

    def getFirstKey(self):
        return next(iter(self.map))

    def getLastKey(self):
        return next(reversed(self.map))

    def clear(self):
        self.map.clear()

class Graph:
    def __init__(self):
        self.data = defaultdict(list)

    def add(self, from_node, to_node):
        self.data[from_node].append(to_node)

    def del_edge(self, from_node, to_node):
        if from_node in self.data and to_node in self.data[from_node]:
            self.data[from_node].remove(to_node)

    def get(self, key):
        return self.data.get(key, [])

    def contains(self, key):
        return key in self.data

    def keySet(self):
        return self.data.keys()

    def isConnect(self, key_1, key_2):
        return key_2 in self.data.get(key_1, [])

    def distList(self, key):
        dist = []
        mark = set()
        stack = [(key, 0)]
        while stack:
            pp, dd = stack.pop()
            if pp in mark:
                continue
            mark.add(pp)
            dist.append(PP(pp, dd))
            for next_node in self.get(pp):
                if next_node not in mark:
                    stack.append((next_node, dd + 1))
        return dist

    def distV(self, key):
        dist = [-1] * (len(self.data) + 1)
        mark = set()
        stack = [(key, 0)]
        while stack:
            pp, dd = stack.pop()
            if pp in mark:
                continue
            mark.add(pp)
            dist[pp] = dd
            for next_node in self.get(pp):
                if next_node not in mark:
                    stack.append((next_node, dd + 1))
        return dist

    def bridgeDfs(self, now, pre):
        self.mark.add(now)
        self.mapCnt[now] = self.number
        self.mapLow[now] = self.number
        low = 0
        for next_node in self.get(now):
            if next_node == pre:
                continue
            if next_node in self.mark:
                if self.mapLow[now] > self.mapLow[next_node]:
                    self.mapLow[now] = self.mapLow[next_node]
                continue
            self.number += 1
            low = self.bridgeDfs(next_node, now)
            if self.mapLow[now] > low:
                self.mapLow[now] = low
        return self.mapLow[now]

    def bridgeCnt(self, start):
        self.mapCnt = {}
        self.mapLow = {}
        self.mark = set()
        self.number = 0
        self.bridgeDfs(start, start)
        ans = 0
        for key in self.mapCnt:
            if self.mapCnt[key] == self.mapLow[key]:
                ans += 1
        return ans - 1

    def dump(self, out_stream):
        for key in self.data:
            out_stream.write(f"{key} : ")
            for val in self.data[key]:
                out_stream.write(f"{val} ")
            out_stream.write("\n")

class GraphWith:
    def __init__(self):
        self.data = defaultdict(list)

    def add(self, key, p):
        self.data[key].append(p)

    def get(self, key):
        return self.data.get(key, [])

    def contains(self, key):
        return key in self.data

    def keySet(self):
        return self.data.keys()

    def isConnect(self, key_1, key_2):
        for p in self.data.get(key_1, []):
            if p.getKey() == key_2:
                return True
        return False

    def distance(self, key_1, key_2):
        mark = set()
        stack = [(key_1, 0)]
        while stack:
            key, val = stack.pop()
            if key in mark:
                continue
            mark.add(key)
            if key == key_2:
                return val
            for pp in self.get(key):
                if pp.getKey() not in mark:
                    stack.append((pp.getKey(), val + pp.getVal()))
        return float('inf')

class GraphLong:
    def __init__(self):
        self.G = defaultdict(list)

    def add(self, key, value):
        self.G[key].append(value)

    def get(self, key):
        return self.G.get(key, [])

class GraphLongWith:
    def __init__(self):
        self.G = defaultdict(list)

    def add(self, key, p):
        self.G[key].append(p)

    def get(self, key):
        return self.G.get(key, [])

class EulerTour:
    def __init__(self):
        self.g = None
        self.euler_tour = []
        self.begin = None
        self.end = None
        self.k = 0
        self.root = 0

    def dfs(self, v, p, out_stream):
        out_stream.write(f"v = {v}  p = {p}\n")
        self.begin[v] = self.k
        self.euler_tour.append(v)
        self.k += 1
        if v not in self.g.data:
            return
        for i in self.g.get(v):
            if i != p:
                self.dfs(i, v, out_stream)
                self.euler_tour.append(v)
                self.k += 1
        self.end[v] = self.k

    def init(self, p_cnt, root, g, out_stream):
        self.begin = [0] * (p_cnt + 1)
        self.end = [0] * (p_cnt + 1)
        self.root = root
        self.g = g
        self.dfs(root, -1, out_stream)

    def getPartTour(self, v):
        return self.euler_tour[self.begin[v]:self.end[v]]

    def getPartList(self, v):
        return list(set(self.getPartTour(v)))

class MyInput:
    def __init__(self, in_stream):
        self.in_stream = in_stream
        self.buffer = []
        self.buffer_pointer = 0

    def read(self):
        if self.buffer_pointer >= len(self.buffer):
            self.buffer = self.in_stream.readline().split()
            self.buffer_pointer = 0
        return self.buffer[self.buffer_pointer]

    def nextInt(self):
        return int(self.read())

    def nextLong(self):
        return int(self.read())

    def nextDouble(self):
        return float(self.read())

    def nextString(self):
        return self.read()

    def nextDChar(self):
        return list(self.read())

    def nextChar(self):
        return self.read()[0]

    def reads(self, len, accept):
        result = []
        while True:
            c = self.read()
            if accept(c):
                break
            result.append(c)
        return result

    class EndOfFileRuntimeException(Exception):
        pass

if __name__ == "__main__":
    solver = Solver(sys.stdin, sys.stdout)
    solver.solve()
# End of Code
